pkg/kubectl/cmd/annotate.go

```diff
@@ -190,10 +190,26 @@ func (o AnnotateOptions) RunAnnotate(f cmdutil.Factory, cmd *cobra.Command) erro
 		return err
 	}
 
+	var includeUninitialized bool
+	if o.all {
+		// include the uninitialized objects by default
+		// unless explicitly set --include-uninitialized=false
+		includeUninitialized = true
+	}
+	if o.selector != "" {
```

Comment:
@caesarxuchao Yes, this is current implementation.

But we have to discuss on such combinations. I think `o.all` is the dominant one. When other flags combine with `o.all`, `includeUninitialized` should be set to `true`.

ping @ahmetb @smarterclayton @pwittrock, PTAL.